internal/controller/controllers/crd_utils.go

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import (
+	"context"
+
+	adiiov1alpha1 "github.com/openshift/assisted-service/internal/controller/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CRDUtils struct {
+	client client.Client
+}
+
+func NewCRDUtils(client client.Client) *CRDUtils {
+	return &CRDUtils{client: client}
+}
+
+func (u *CRDUtils) CreateAgentCR(ctx context.Context, hostId, clusterNamespace, clusterName string) error {
+	hostCr := &adiiov1alpha1.Agent{
+		Spec: adiiov1alpha1.AgentSpec{
+			ClusterDeploymentName: &adiiov1alpha1.ClusterReference{
+				Name:      clusterName,
+				Namespace: clusterNamespace,
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Agent",
+			APIVersion: "adi.io.my.domain/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      hostId,
+			Namespace: clusterNamespace,
+		},
+	}
+	return u.client.Create(ctx, hostCr)
```

Comment:
What will happen if the host already exist?
Wrong boot order, unexpected reset etc.

Reply:
Right, we will get an error:
`"reason":"agents.adi.io.my.domain \"3fa85f64-5717-4562-b3fc-2c963f66afa6\" already exists"`

I will need to check before creating.

Reply:
Probably yes, it's ok if host is registered twice. we will just need to reconcile the host again, so later you can use an event instead of creating it again.

Reply:
Done